src/compiler/parser/index.js

```diff
@@ -761,6 +762,12 @@ function processAttrs (el) {
   for (i = 0, l = list.length; i < l; i++) {
     name = rawName = list[i].name
     value = list[i].value
+    // :v-if or similar
+    if (process.env.NODE_ENV !== 'production' && colonDirRE.test(name)) {
+      warn(
+        `A v-bind shorthand directive was used on another Vue directive. Did you want to write '${name.substr(1)}="${value}"'?`
```

Comment:
I think we could directly say which directive was used instead, something like:

```suggestion
        `A v-bind shorthand was used on another Vue Directive: ${name}. Did you meant '${name.substr(1)}="${value}"'?`
```

Reply:
Makes sense! It might look a bit weird, though, given that `${name}` includes an initial colon:

> A v-bind shorthand was used on another Vue Directive: :v-if. Did you mean 'v-if="foo"'

what about going full explicit?

> A v-bind shorthand was used on a v-if directive. Did you mean v-if="foo", without a colon?




Thanks for the feedback! 🙌

Reply:
you are right, let's be even more explicit, I updated the suggestion and fixed the typos too

Reply:
Done! Didn't commit the suggestion since I needed to update the test as well 👍